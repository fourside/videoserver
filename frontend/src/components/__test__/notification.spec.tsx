import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import Notification from '../notification';

test('snapshot', () => {
  const component = renderer
    .create(<Notification message="OK" isShown={true} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('message', () => {
  const component = shallow(<Notification message="OK" isShown={true} />);
  expect(component.find('span').text()).toBe('OK');
});

test('isShown is true', () => {
  const component = shallow(<Notification message="OK" isShown={true} />);
  expect(component.hasClass('is-shown')).toBe(true);
});

test('isShown is false', () => {
  const component = shallow(<Notification message="OK" isShown={false} />);
  expect(component.hasClass('is-hidden')).toBe(true);
});
