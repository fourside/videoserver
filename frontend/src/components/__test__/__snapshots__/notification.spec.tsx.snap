// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="notification is-success http-notify is-shown"
>
  <button
    className="delete"
  />
  <span>
    OK
  </span>
</div>
`;
